lib/diagnostic.rb

```diff
@@ -43,6 +44,10 @@ def transform(words)
   # replacing the trailing newline character with a
   # space and return the result.
   def read_file(filename)
-    # Your code goes here
+      lines = File.open(filename[0]) do |file|
+      file.each_with_object(lines: 0, new_lines: 0) do |line, memo|
+        memo[:lines] += line.split()
+        memo[:new_lines] += line.join(' ')
+      end
```

Comment:
Here's one possible solution:
`File.open(filename).map{ | line | line.chomp}.join(' ').reduce(&:+).strip`
The strip at the end is to get rid of any trailing whitespace we may have accidentally included.